MaterialDesignThemes.Wpf/NumericUpDown.cs

```diff
@@ -0,0 +1,257 @@
+
+    #region DependencyProperty : AllowChangeOnScroll
+
+    public bool AllowChangeOnScroll
+    {
+        get => (bool)GetValue(AllowChangeOnScrollProperty);
+        set => SetValue(AllowChangeOnScrollProperty, value);
+    }
+
+    public static readonly DependencyProperty AllowChangeOnScrollProperty =
+        DependencyProperty.Register(nameof(AllowChangeOnScroll), typeof(bool), typeof(NumericUpDown), new PropertyMetadata(false));
+
+    #endregion
+
+    #endregion DependencyProperties
+
+    #region Event : ValueChangedEvent
+    [Category("Behavior")]
+    public static readonly RoutedEvent ValueChangedEvent = EventManager.RegisterRoutedEvent(nameof(ValueChanged), RoutingStrategy.Bubble, typeof(RoutedPropertyChangedEventHandler<int>), typeof(NumericUpDown));
+
+    public event RoutedPropertyChangedEventHandler<int> ValueChanged
+    {
+        add => AddHandler(ValueChangedEvent, value);
+        remove => RemoveHandler(ValueChangedEvent, value);
+    }
+    #endregion Event : ValueChangedEvent
+
+    public override void OnApplyTemplate()
+    {
+        if (_increaseButton != null)
+            _increaseButton.Click -= IncreaseButtonOnClick;
+
+        if (_decreaseButton != null)
+            _decreaseButton.Click -= DecreaseButtonOnClick;
+        if (_textBoxField != null)
+            _textBoxField.TextChanged -= OnTextBoxFocusLost;
+
+        _increaseButton = GetTemplateChild(IncreaseButtonPartName) as RepeatButton;
+        _decreaseButton = GetTemplateChild(DecreaseButtonPartName) as RepeatButton;
+        _textBoxField = GetTemplateChild(TextFieldBoxPartName) as TextBox;
+
+        if (_increaseButton != null)
+            _increaseButton.Click += IncreaseButtonOnClick;
+
+        if (_decreaseButton != null)
+            _decreaseButton.Click += DecreaseButtonOnClick;
+
+        if (_textBoxField != null)
+        {
+            _textBoxField.LostFocus += OnTextBoxFocusLost;
+            _textBoxField.Text = Value.ToString(CultureInfo.CurrentUICulture);
+        }
+
+        base.OnApplyTemplate();
+    }
+
+    private void OnTextBoxFocusLost(object sender, EventArgs e)
+    {
+        if (_textBoxField is { } textBoxField)
+        {
+            if (int.TryParse(textBoxField.Text, NumberStyles.Integer, CultureInfo.CurrentUICulture, out int numericValue))
+            {
+                SetCurrentValue(ValueProperty, numericValue);
+            }
+            else
+            {
+                textBoxField.Text = Value.ToString(CultureInfo.CurrentUICulture);
+            }
+        }
+    }
+
+    private void IncreaseButtonOnClick(object sender, RoutedEventArgs e) => OnIncrease();
+
+    private void DecreaseButtonOnClick(object sender, RoutedEventArgs e) => OnDecrease();
+
+    private void OnIncrease()
+    {
+        SetCurrentValue(ValueProperty, Value + 1);
+
+        if (IncreaseCommand?.CanExecute(IncreaseCommandParameter) ?? false)
+            IncreaseCommand.Execute(IncreaseCommandParameter);
```

Comment:
Related to my comment above, this feels weird to me. The command should be used to actually change the value, not be executed once the value has changed. For that, there is a `ValueChanged` `RoutedEvent`.